astropy/tests/helper.py

```diff
@@ -604,3 +604,40 @@ def check_pickling_recovery(original, protocol):
     class_history = [original.__class__]
     generic_recursive_equality_test(original, unpickled,
                                     class_history)
+
+
+def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=0, err_msg='', verbose=True):
+    """
+    Raise an assertion if two objects are not equal up to desired tolerance.
+
+    This is a :class:`~astropy.units.Quantity`-aware version of
+    :func:`numpy.testing.assert_allclose`.
+    """
+
+    import numpy as np
+    from ..units import Quantity
+
+    if isinstance(actual, Quantity) and isinstance(desired, Quantity):
```

Comment:
I feel this is excessive testing of inputs! Elsewhere, anything not a `Quantity` is treated as dimensionless, and I would just do that here too, raising `UnitsError` if necessary. So, I would do:
```
actual = Quantity(actual, subok=True, copy=False)
desired = Quantity(desired, subok=True, copy=False).to(actual.unit)
atol = Quantity(atol, subok=True, copy=False).to(actual.unit)
rtol=  Quantity(rtol, subok=True, copy=False).to(u.dimensionless_unscaled)
np.testing.assert_allclose(actual.value, desired.value,
                           rtol=rtol.value, atol=atol.value, err_msg=err_msg, verbose=verbose)
```